### ISSUE TITLE
Spawning processes with `:!` fails on gvim

### ISSUE THREAD

**Issue by felipec** [vim/vim#8314]
When I launch a background process with gvim and shelltemp and zsh, the spawn process exits immediately and nothing happens:

```sh
SHELL=zsh vim -g -c 'set shelltemp' -c ':!touch /tmp/foobar &'
```

This is a problem when trying to open a browser with `gx` and using `xdg-open` (when it uses `exo-open`), see issue #4738 for more context.

The following patch fixes the problem:

```diff
--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -4775,11 +4775,6 @@ mch_call_shell_fork(
                    // push stream discipline modules
                    if (options & SHELL_COOKED)
                        setup_slavepty(pty_slave_fd);
-#  ifdef TIOCSCTTY
-                   // Try to become controlling tty (probably doesn't work,
-                   // unless run by root)
-                   ioctl(pty_slave_fd, TIOCSCTTY, (char *)NULL);
-#  endif
                }
 # endif
                set_default_child_environment(FALSE);
```

**Environment (please complete the following information):**
 - Vim 8.2.2891
 - OS: Arch Linux
 - Terminal: GUI

**Comment by lacygoill** [comment]
Fixed by [8.2.2919](https://github.com/vim/vim/releases/tag/v8.2.2919)

**Comment by felipec** [comment]
Nice! FWIW the actual commit is 6a43b37b7.

I guess this can be closed, but it still would good to try to understand how to reproduce the issue.
